### PR TITLE
guide: Azure and GCP remote pages

### PR DESCRIPTION
**Pull request by jorgeorpinel**
To complete the remote guides where cloud versioning apply.

In review app:

- https://dvc-org-guide-data-mgmt-k1tu8s.herokuapp.com/doc/user-guide/data-management/remote-storage/azure-blob-storage
- https://dvc-org-guide-data-mgmt-k1tu8s.herokuapp.com/doc/user-guide/data-management/remote-storage/google-cloud-storage

---

- [x] Do not merge before parent PR, https://github.com/iterative/dvc.org/pull/4264
- [ ] Next: Google Drive? See https://github.com/iterative/dvc.org/pull/4284#discussion_r1094265986 below.

### CODE REVIEW

File: content/docs/user-guide/data-management/remote-storage/amazon-s3.md [hunk 1]
```diff
@@ -1,21 +1,25 @@
 # Amazon S3
 
+<!--
+## Amazon S3
+-->
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/remote-storage/amazon-s3.md:5]: Some small updates here to match the format improvements in the new pages.

File: content/docs/sidebar.json [hunk 2]
```diff
@@ -127,7 +127,11 @@
           {
             "slug": "remote-storage",
             "source": "remote-storage/index.md",
-            "children": ["amazon-s3"]
+            "children": [
+              "amazon-s3",
+              "azure-blob-storage",
+              "google-cloud-storage"
```
Comment (jorgeorpinel) [inline content/docs/sidebar.json:133]: Added these 2!

File: content/docs/user-guide/data-management/cloud-versioning.md [hunk 3]
```diff
@@ -50,15 +50,21 @@ remote storage performance optimizations will be unavailable.
...
-- Microsoft Azure Blob Storage (requires
-  [Blob versioning](https://learn.microsoft.com/en-us/azure/storage/blobs/versioning-overview)
-  enabled storage accounts and containers)
-- Google Cloud Storage (requires
-  [Object versioning](https://cloud.google.com/storage/docs/object-versioning)
-  enabled buckets)
+- [Amazon S3] (requires [S3 Versioning] enabled buckets)
+- Microsoft [Azure Blob Storage] (requires [Blob versioning] enabled storage
+  accounts and containers)
+- [Google Cloud Storage] (requires [Object versioning] enabled buckets)
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/cloud-versioning.md:56]: Updated these links.

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
+# Microsoft Azure Blob Storage
+
+<!--
+## Microsoft Azure Blob Storage
+-->
+
+`dvc remote add` a (new) remote name and a valid [Azure Blob Storage] URL:
+
+```cli
+$ dvc remote add -d myremote azure://<mycontainer>/<path>
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:10]: All these details are now removed from the `remote add/modify` refs.

File: content/docs/user-guide/data-management/remote-storage/google-cloud-storage.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
+# Google Cloud Storage
+
+<!--
+## Google Cloud Storage
+-->
+
+`dvc remote add` a (new) remote name and a valid [Google Cloud Storage] URL:
+
+```cli
+$ dvc remote add -d myremote gs://<mybucket>/<path>
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/remote-storage/google-cloud-storage.md:10]: Same here

File: content/docs/user-guide/how-to/setup-google-drive-remote.md [hunk 6]
```diff
@@ -269,8 +269,8 @@ heavy usage, it is recommended to rely on
 </admon>
 
 > This requires having your own
-> [GC project](/doc/user-guide/how-to/setup-google-drive-remote#using-a-custom-google-cloud-project-recommended)
-> as explained above.
+> [GC project](#using-a-custom-google-cloud-project-recommended) as explained
+> above.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/how-to/setup-google-drive-remote.md:273]: Finally, just a quick internal link fix here for now, but this page ([GDrive](https://dvc.org/doc/user-guide/how-to/setup-google-drive-remote#how-to-setup-a-google-drive-dvc-remote)) would probably be next (turn it into a regular remote config page). Although maybe Aliyun OSS is easier (similar to the ones that are done)... 🤔

**Review by jorgeorpinel — COMMENTED** [review]
Change summary:

**Comment by jorgeorpinel** [comment]
@dberenbaum thanks for taking a look earlier. Yup this was waiting for https://github.com/iterative/dvc.org/pull/4264, which seems close to ready. I solved conflicts and applied other changes to the template from that PR to these new pages so this one should also be readyish. PTAL when you can.

> I'll try to quickly contribute the rest of types by copy/pasting this template afterwards in an effort to close #2866. Some may need being taken over by your team though. No rush though, I'll be around 🙂

**Comment by jorgeorpinel** [comment]
Fixed conflicts again 👍🏼

File: content/docs/user-guide/data-management/remote-storage/index.md [hunk 7]
```diff
@@ -92,27 +92,29 @@ team.
...
+- [Google Cloud Storage] (GCP)
 - Aliyun OSS
 
+<admon type="tip">
+
+Guides for each storage type are in progress. For storage types that do not link
+to a specific guide, see the
+[`dvc remote modify` reference](/doc/command-reference/remote/modify#supported-storage-types).
+
+</admon>
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/remote-storage/index.md:109]: Oops didn't mean to move this under the list. Somehow happened during Git merge.

But how about we remove this admon and link every entry to the corresponding `remote modify` section instead for now?

Comment (jorgeorpinel) [reply content/docs/user-guide/data-management/remote-storage/index.md]: > p.s. moved the admon up again for now.

Comment (dberenbaum) [reply content/docs/user-guide/data-management/remote-storage/index.md]: TBH I don't have a strong opinion. If you prefer that, feel free to update to that format.

File: content/docs/user-guide/data-management/remote-storage/amazon-s3.md [hunk 8]
```diff
@@ -79,7 +79,7 @@ $ dvc remote modify --local myremote \
 $ dvc remote modify --local myremote \
                     credentialpath 'path/to/credentials'
 # and (optional)
-$ dvc remote modify --local myremote profile 'myprofile'
+$ dvc remote modify --local myremote profile 'mysecret'
```
Comment (dberenbaum) [inline content/docs/user-guide/data-management/remote-storage/amazon-s3.md:82]: Why this change? A profile name shouldn't be a secret.

**Review by dberenbaum — APPROVED** [review]

File: content/docs/user-guide/data-management/remote-storage/index.md (hunk 7 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/data-management/remote-storage/index.md]: Well, for now it's not easy for me to make more changes here since I can't push to the branch anymore but I could open a separate PR from my fork later. Let's keep as-is in this one for now.

File: content/docs/user-guide/data-management/remote-storage/amazon-s3.md (hunk 8 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/data-management/remote-storage/amazon-s3.md:82]: If it's not a secret we don't need `--local` either. Up to you:

```suggestion
$ dvc remote modify myremote profile 'myprofile'
```

File: content/docs/user-guide/data-management/remote-storage/index.md (hunk 7 above)
Comment (shcheklein) [reply content/docs/user-guide/data-management/remote-storage/index.md]: @jorgeorpinel hey, I invited you back. sorry about that, was not intentional, I don't think we need complicating things here.

Comment (jorgeorpinel) [reply content/docs/user-guide/data-management/remote-storage/index.md]: No worries I assumed it had to do with compliance. I don't mind either way. Thanks for the invite back BTW I was auto-subscribed to 165 repos heh. Any chance I can only be added to this repo instead of the teams to un-do that?

Comment (shcheklein) [reply content/docs/user-guide/data-management/remote-storage/index.md]: yes, sorry about that. I can do that. CML and DVC I guess - just in case.

Comment (shcheklein) [reply content/docs/user-guide/data-management/remote-storage/index.md]: done, you should have the maintainer access to a few repos now

Comment (jorgeorpinel) [reply content/docs/user-guide/data-management/remote-storage/index.md]: Thanks @shcheklein ! OK I linked all the types now. PTAL @dberenbaum 🙂

File: content/docs/user-guide/data-management/remote-storage/amazon-s3.md (hunk 8 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/data-management/remote-storage/amazon-s3.md]: Sure, I would do that. But I believe in general the recommendation is to be extra paranoid and don't even share that info. Anyway, 👍🏼
